### PR TITLE
[RFC] vim-patch: 0c1ff16, 939a1ab, 8067a64

### PR DESCRIPTION
**Pull request by Shougo**
```
vim-patch:0c1ff16

updated runtime files.  Add avra syntax.

https://github.com/vim/vim/commit/0c1ff16b5467f97ce08134fdbc8198127bbe492a

vim-patch:939a1ab

Updated runtime files.

https://github.com/vim/vim/commit/939a1abe935a539f2d4c90a56cb0682cbaf3bbb0

vim-patch:8067a64

Add missing test file.

https://github.com/vim/vim/commit/8067a64852d6d134b493c5674e404225ed4bbe7d
```

### CODE REVIEW

File: runtime/syntax/vim.vim [hunk 1]
```diff
@@ -17,47 +17,126 @@ set cpo&vim
...
+syn keyword vimFuncName contained	alloc_fail argidx assert_equal assert_match browse bufloaded byte2line ceil ch_evalexpr ch_info ch_read ch_setoptions col confirm count delete disable_char_avail_for_testing eventhandler 
+
+"--- syntax here and above generated by mkvimvim ---
 " Special Vim Highlighting (not automatic) {{{1
```
Comment (justinmk) [inline runtime/syntax/vim.vim]: Everything above this line is auto-generated at build-time, for Neovim. I discarded above changes in the merge-commit.

I know this is confusing, we need to improve this.

File: src/nvim/testdir/test_ex_undo.vim [hunk 2]
```diff
@@ -0,0 +1,19 @@
+" Tests for :undo
```
Comment (justinmk) [inline src/nvim/testdir/test_ex_undo.vim]: This also needs to be added to `test_alot.vim`. Done in the merge-commit.

File: runtime/syntax/vim.vim [hunk 3]
```diff
@@ -301,10 +380,8 @@ syn case match
 " ====
 syn match	vimMap		"\<map\>!\=\ze\s*[^(]" skipwhite nextgroup=vimMapMod,vimMapLhs
 syn keyword	vimMap		cm[ap] cno[remap] im[ap] ino[remap] lm[ap] ln[oremap] nm[ap] nn[oremap] no[remap] om[ap] ono[remap] smap snor[emap] vm[ap] vn[oremap] xm[ap] xn[oremap] skipwhite nextgroup=vimMapBang,vimMapMod,vimMapLhs
-syn keyword	nvimMap		tn[oremap] tm[ap] skipwhite nextgroup=vimMapBang,vimMapMod,vimMapLhs
 syn keyword	vimMap		mapc[lear] smapc[lear]
 syn keyword	vimUnmap		cu[nmap] iu[nmap] lu[nmap] nun[map] ou[nmap] sunm[ap] unm[ap] unm[ap] vu[nmap] xu[nmap] skipwhite nextgroup=vimMapBang,vimMapMod,vimMapLhs
-syn keyword	nvimUnmap 		tunm[ap] skipwhite nextgroup=vimMapBang,vimMapMod,vimMapLhs
```
Comment (justinmk) [inline runtime/syntax/vim.vim]: These `nvimFoo` groups should not be deleted. Fixed in the merge-commit.

**Comment by justinmk** [comment]
Merging `syntax/vim.vim` is really error-prone. We need to fix that...

**Comment by Shougo** [comment]
Thank you for the merging!
I get the your fix.

**Pull request merged**